src/test/java/seedu/address/logic/commands/DeleteStudentCommandTest.java

```diff
@@ -5,58 +5,152 @@
+    }
+
+    // Ensure CommandException is thrown when duplicate student names exist, and only name is provided
+    @Test
+    public void execute_duplicateStudentNames_throwsCommandException() {
+        Student validStudent1 = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        Student validStudent2 = new StudentBuilder().withName("John Ng").withStudentNumber("A0123456Y").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent1, validStudent2);
+
+        DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"));
+
+        assertThrows(CommandException.class, "There is more than 1 student of the same name.\n"
+                + "Their student numbers are as follows: A1234567X A0123456Y " + "\n"
+                + "Use the following command: deletes n/John Ng "
+                + "sn/STUDENT NUMBER to delete the student.", () ->
+                command.execute(modelStub));
+    }
+
+    // Ensure deletion by only provided name works when student with correct details exists
     @Test
-    public void execute_studentDelete_success() throws Exception {
+    public void execute_singleStudentDeleteByNameOnly_success() throws Exception {
         Student validStudent = new StudentBuilder().withName("John Ng").build();
         DeleteStudentCommandTest.ModelStubWithStudent modelStub =
                 new DeleteStudentCommandTest.ModelStubWithStudent(validStudent);
-        modelStub.addStudent(validStudent);
 
         DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"));
 
         CommandResult result = command.execute(modelStub);
 
-        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()),
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()
+                + " " + validStudent.getStudentNumber()),
                 result.getFeedbackToUser());
     }
 
+    // Ensure deletion by provided name and provided student number works when student with correct details exists
+    @Test
+    public void execute_singleStudentDeleteByNameAndStudentNumber_success() throws Exception {
+        Student validStudent = new StudentBuilder().withName("John Ng").withStudentNumber("A1234567X").build();
+        DeleteStudentCommandTest.ModelStubWithStudent modelStub =
+                new DeleteStudentCommandTest.ModelStubWithStudent(validStudent);
+
+        DeleteStudentCommand command = new DeleteStudentCommand(new Name("John Ng"), new StudentNumber("A1234567X"));
+
+        CommandResult result = command.execute(modelStub);
+
+        assertEquals(String.format(DeleteStudentCommand.MESSAGE_DELETE_STUDENT_SUCCESS + validStudent.getName()
+                        + " " + validStudent.getStudentNumber()),
+                result.getFeedbackToUser());
```

Comment:
You can consider using the assertCommandSuccess or something in the CommandUtils i think.